### PR TITLE
fix: updated the PR- feat : Implemented details on successful transaction

### PR DESCRIPTION
**Pull request by alphaNewrex**
## Issue Fix
Fixes merge issues in PR #428 
and Fixes Issue #425

## Description
Added a transaction receipt when a transaction is successful. this can be used to send the screenshot of the amount of money sent to anyone in order to confirm the money dispatched from our side.

##

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by alphaNewrex** [comment]
@devansh-299 look into this PR and please review it.

File: mifospay/src/main/res/layout/fragment_make_transfer.xml [hunk 1]
```diff
@@ -110,15 +110,67 @@
...
+            android:visibility="gone">
+
+            <View
+                android:id="@+id/view_transfer_success"
+                android:layout_width="@dimen/value_80dp"
+                android:layout_height="@dimen/value_80dp"
+                android:layout_gravity="center"
+                android:layout_marginBottom="@dimen/value_20dp"
+                android:layout_marginTop="@dimen/value_20dp"
+                android:layout_marginRight="@dimen/value_20dp"
```
Comment (devansh-299) [inline mifospay/src/main/res/layout/fragment_make_transfer.xml]: To support possible use of RTL in future, better use `marginEnd` instead

File: mifospay/src/main/res/layout/fragment_make_transfer.xml [hunk 2]
```diff
@@ -110,15 +110,67 @@
...
+                android:background="@drawable/transfer_success"
+                android:layout_alignParentRight="true"
```
Comment (devansh-299) [inline mifospay/src/main/res/layout/fragment_make_transfer.xml]: same here

File: mifospay/src/main/res/layout/fragment_make_transfer.xml [hunk 3]
```diff
@@ -110,15 +110,67 @@
...
+                android:visibility="visible"/>
```
Comment (devansh-299) [inline mifospay/src/main/res/layout/fragment_make_transfer.xml]: space between `"` and `/>` is missing.

File: mifospay/src/main/res/layout/fragment_make_transfer.xml [hunk 4]
```diff
@@ -110,15 +110,67 @@
...
+            <TextView
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:id="@+id/payment_success"
+                android:layout_toLeftOf="@id/view_transfer_success"
```
Comment (devansh-299) [inline mifospay/src/main/res/layout/fragment_make_transfer.xml]: Same here, use `toStartOf` instead

**Review by devansh-299 — CHANGES_REQUESTED** [review]
Few more changes:
* Code formatting has several issues, just use _Control+Alt+L_ (Windows) or _Command+Option+L_ (MacOS) to reformat the code
* View  ids in the XML file are not meaningful. Follow [this article](https://medium.com/mindorks/android-resource-naming-convention-42e4e8026614) while naming views

File: mifospay/src/main/res/layout/fragment_make_transfer.xml [hunk 5]
```diff
@@ -110,15 +110,67 @@
...
+                android:layout_marginBottom="@dimen/value_20dp"
+                android:layout_marginTop="@dimen/value_20dp"
+                android:layout_marginRight="@dimen/value_20dp"
+                android:background="@drawable/transfer_success"
+                android:layout_alignParentRight="true"
+                android:visibility="visible"/>
+            <TextView
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:id="@+id/payment_success"
```
Comment (devansh-299) [inline mifospay/src/main/res/layout/fragment_make_transfer.xml]: View Id ideally should tell the use of the view, here it's not related to the its purpose

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/common/ui/MakeTransferFragment.java [hunk 6]
```diff
@@ -74,12 +79,29 @@ public class MakeTransferFragment extends BottomSheetDialogFragment
...
+    TextView receiverName;
+
+    @BindView(R.id.transction_id)
+    TextView transactionId;
+
+    @BindView(R.id.show_receipt)
+    Button showReceipt;
+
+    @BindView(R.id.layout_success)
+    RelativeLayout relativeLayout;
```
Comment (devansh-299) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/common/ui/MakeTransferFragment.java]: Use `successLayout` as the variable name

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/common/ui/MakeTransferFragment.java [hunk 7]
```diff
@@ -156,6 +179,23 @@ public void showToClientDetails(long clientId, String name, String externalId) {
 
     @Override
     public void transferSuccess() {
+        relativeLayout.setVisibility(View.VISIBLE);
+        Transaction transaction;
```
Comment (devansh-299) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/common/ui/MakeTransferFragment.java]: The variable declaration can be done in the same line as the assignment

**Review by devansh-299 — CHANGES_REQUESTED** [review]

**Comment by devansh-299** [comment]
> @devansh-299 look into this PR and please review it.

@alphaNewrex UI is not up to the mark, please make the following changes:
* The heading text should be changed to `Transaction Successful`
* The views should be arranged in the following verticle fashion: 
    1. Transaction ID as `Transaction Id: <transaction id>`
    2. The receiver as `Transferred To: <receiver name>`
    3. Amount as `Amount: <currency><amount>`

File: mifospay/src/main/res/layout/fragment_make_transfer.xml [hunk 8]
```diff
@@ -110,15 +110,67 @@
...
+                android:layout_marginTop="@dimen/value_10dp"
+                android:textColor="@color/black"
+                android:layout_below="@id/receiver_name"
+                />
+            <Button
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:id="@+id/show_receipt"
+                android:text="@string/view_receipt"
+                android:textColor="@color/primaryDarkBlue"
```
Comment (devansh-299) [inline mifospay/src/main/res/layout/fragment_make_transfer.xml]: Button text won't be visible, use white instead

File: mifospay/src/main/res/layout/fragment_make_transfer.xml [hunk 9]
```diff
@@ -110,15 +110,67 @@
...
+                android:layout_below="@+id/transction_id"
+                android:layout_marginTop="@dimen/value_10dp"
```
Comment (devansh-299) [inline mifospay/src/main/res/layout/fragment_make_transfer.xml]: Increase the margin to 40dp

File: mifospay/src/main/res/layout/fragment_make_transfer.xml [hunk 10]
```diff
@@ -110,15 +110,67 @@
...
-            android:layout_height="130dp"
-            android:layout_gravity="center"
-            android:layout_marginBottom="@dimen/value_20dp"
-            android:layout_marginTop="@dimen/value_20dp"
-            android:background="@drawable/transfer_success"
-            android:visibility="gone"/>
+        <RelativeLayout
+            android:id="@+id/layout_success"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
```
Comment (devansh-299) [inline mifospay/src/main/res/layout/fragment_make_transfer.xml:120]: Use `wrap_content` instead

File: mifospay/src/main/res/layout/fragment_make_transfer.xml [hunk 11]
```diff
@@ -110,15 +110,67 @@
...
+            android:visibility="gone">
+
+            <View
+                android:id="@+id/view_transfer_success"
+                android:layout_width="@dimen/value_80dp"
+                android:layout_height="@dimen/value_80dp"
+                android:layout_gravity="center"
```
Comment (devansh-299) [inline mifospay/src/main/res/layout/fragment_make_transfer.xml]: No need for this

File: mifospay/src/main/res/layout/fragment_make_transfer.xml [hunk 12]
```diff
@@ -110,15 +110,67 @@
...
+                android:layout_marginBottom="@dimen/value_20dp"
+                android:layout_marginTop="@dimen/value_20dp"
```
Comment (devansh-299) [inline mifospay/src/main/res/layout/fragment_make_transfer.xml]: A better way of placing this will be to vertically centralize this view

**Review by devansh-299 — CHANGES_REQUESTED** [review]

**Comment by alphaNewrex** [comment]
@devansh-299 I have updated the

 UI and made it Align Vertical. A sample for the same is attached below. Since the demo account credentials didn't have any transactions associated with them. I'm attaching a screenshot from Design Tab. The Text for each will be updated by the code.
Screenshot 2022-04-15 013228

**Pull request closed without merging**